poglink/cogs/rates.py

```diff
@@ -108,71 +113,73 @@ async def send_embed(self, description, url):
             logger.info("Announcement channel detected: Publishing message")
             await message.publish()
 
+    async def compare_posted_rates(self, webpage_url, output_path):
```

Comment:
This function is basically what used to be written inside the main `while True` loop you had before. Same thing, just split out for easier testability.